panic when import `golang.org/x/text`

The following program `sample.go` triggers a panic:

```go
package main

import (
	"golang.org/x/text/language"
	"log"
)

func main() {
	log.Println(language.English)
}

```

Expected result:
```console
$ go run ./sample.go

2021/06/19 21:42:46 en
```

Got: `panic run: language: tag is not well-formed`
 Links to panic line number
- [golang.org/x/text/internal/language/tags.go:12](https://github.com/golang/text/blob/master/internal/language/tags.go#L12)
- [golang.org/x/text/internal/language/compact/compact.go:59](https://github.com/golang/text/blob/master/internal/language/compact/compact.go#L59)

```console
$ yaegi ./sample.go


/Users/wlun/go/src/golang.org/x/text/internal/language/tags.go:12:3: panic
/Users/wlun/go/src/golang.org/x/text/internal/language/compact/compact.go:59:20: panic
run: language: tag is not well-formed
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc0013c6800, 0x40)
	/usr/local/Cellar/go/1.16.3/libexec/src/runtime/debug/stack.go:24 +0x9f
github.com/traefik/yaegi/interp.(*Interpreter).eval.func1(0xc00223fc18)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/interp.go:503 +0xc8
panic(0x1b3a800, 0xc001a3dc20)
	/usr/local/Cellar/go/1.16.3/libexec/src/runtime/panic.go:965 +0x1b9
github.com/traefik/yaegi/interp.runCfg.func1(0xc001bccfd0, 0xc002052100, 0xc00223de68)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/run.go:185 +0x253
panic(0x1b3a800, 0xc001a3dc20)
	/usr/local/Cellar/go/1.16.3/libexec/src/runtime/panic.go:965 +0x1b9
github.com/traefik/yaegi/interp.runCfg.func1(0xc001bcd130, 0xc001295900, 0xc00223dc58)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/run.go:185 +0x253
panic(0x1b3a800, 0xc001a3dc20)
	/usr/local/Cellar/go/1.16.3/libexec/src/runtime/panic.go:965 +0x1b9
github.com/traefik/yaegi/interp._panic.func1(0xc001bcd130, 0xc0018fa550)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/run.go:838 +0x7f
github.com/traefik/yaegi/interp.runCfg(0xc001295900, 0xc001bcd130)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/run.go:191 +0x87
github.com/traefik/yaegi/interp.call.func7(0xc001bccfd0, 0xc0015b16c0)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/run.go:1272 +0xd05
github.com/traefik/yaegi/interp.runCfg(0xc002052100, 0xc001bccfd0)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/run.go:191 +0x87
github.com/traefik/yaegi/interp.(*Interpreter).run(0xc000366000, 0xc001ea5800, 0xc000368000)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/run.go:122 +0x2b0
github.com/traefik/yaegi/interp.(*Interpreter).importSrc(0xc000366000, 0x0, 0x0, 0xc001d0ac91, 0x2b, 0x1078e01, 0xc0004a6158, 0x8, 0x0, 0x0)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/src.go:156 +0xad9
github.com/traefik/yaegi/interp.(*Interpreter).gta.func1(0xc0014ffd00, 0xc00223ec48)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/gta.go:222 +0x12e5
github.com/traefik/yaegi/interp.(*node).Walk(0xc0014ffd00, 0xc00223ec48, 0x0)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/interp.go:232 +0xb5
github.com/traefik/yaegi/interp.(*node).Walk(0xc0014fef00, 0xc00223ec48, 0x0)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/interp.go:236 +0x66
github.com/traefik/yaegi/interp.(*node).Walk(0xc0014fe800, 0xc00223ec48, 0x0)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/interp.go:236 +0x66
github.com/traefik/yaegi/interp.(*Interpreter).gta(0xc000366000, 0xc0014fe800, 0xc00011e580, 0x1a, 0xc000488041, 0x1a, 0xc001bbfc28, 0x8, 0xc0014fe800, 0x0, ...)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/gta.go:20 +0x21f
github.com/traefik/yaegi/interp.(*Interpreter).importSrc(0xc000366000, 0x0, 0x0, 0xc000488041, 0x1a, 0x1, 0x16, 0xc000211300, 0xca, 0xc000212510)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/src.go:100 +0xf57
github.com/traefik/yaegi/interp.(*Interpreter).gta.func1(0xc000210400, 0x0)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/gta.go:222 +0x12e5
github.com/traefik/yaegi/interp.(*node).Walk(0xc000210400, 0xc00223f9b8, 0x0)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/interp.go:232 +0xb5
github.com/traefik/yaegi/interp.(*node).Walk(0xc000210300, 0xc00223f9b8, 0x0)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/interp.go:236 +0x66
github.com/traefik/yaegi/interp.(*node).Walk(0xc000210100, 0xc00223f9b8, 0x0)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/interp.go:236 +0x66
github.com/traefik/yaegi/interp.(*Interpreter).gta(0xc000366000, 0xc000210100, 0xc00020d9d0, 0x4, 0xc00020d9d0, 0x4, 0x10, 0xc000200240, 0x0, 0xc0002063e0, ...)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/gta.go:20 +0x21f
github.com/traefik/yaegi/interp.(*Interpreter).gtaRetry(0xc000366000, 0xc00223fb88, 0x1, 0x1, 0xc00020d9d0, 0x4, 0xc00020d9d0, 0x4)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/gta.go:311 +0x12c
github.com/traefik/yaegi/interp.(*Interpreter).eval(0xc000366000, 0xc0002c6310, 0x6e, 0x7ffeefbffb2e, 0x9, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/interp.go:525 +0x2a5
github.com/traefik/yaegi/interp.(*Interpreter).EvalPath(0xc000366000, 0x7ffeefbffb2e, 0x9, 0xc0002c6200, 0x6e, 0x0, 0x0, 0xc0001b1cb0)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/interp/interp.go:417 +0x105
main.runFile(0xc000366000, 0x7ffeefbffb2e, 0x9, 0x0, 0x1, 0x0)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/cmd/yaegi/run.go:139 +0xb1
main.run(0xc000032050, 0x1, 0x1, 0xc00000a501, 0xc000001680)
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/cmd/yaegi/run.go:102 +0x99d
main.main()
	/Users/wlun/go/pkg/mod/github.com/traefik/yaegi@v0.9.18/cmd/yaegi/yaegi.go:144 +0x415
```

--- a/_test/alias3.go
+++ b/_test/alias3.go
@@ -17,3 +17,6 @@ func (t *T) PrintT() {
 func main() {
 	globalT.PrintT()
 }
+
+// Output:
+// test

--- a/_test/alias3/alias3.go
+++ b/_test/alias3/alias3.go
@@ -7,6 +7,3 @@ type T struct {
 func (t *T) Print() {
 	println(t.A)
 }
-
-// Output:
-// test

--- a/_test/selector-scope0.go
+++ b/_test/selector-scope0.go
@@ -9,10 +9,20 @@ func test(time string, t time.Time) string {
 	return time
 }
 
+var zero = time.Time{}
+
+func test2(time string) time.Time {
+	return zero
+}
+
 func main() {
 	str := test("test", time.Now())
 	fmt.Println(str)
+
+	str2 := test2("test2")
+	fmt.Println(str2)
 }
 
 // Output:
 // test
+// 0001-01-01 00:00:00 +0000 UTC

--- a/interp/type.go
+++ b/interp/type.go
@@ -777,28 +777,17 @@ func nodeType2(interp *Interpreter, sc *scope, n *node, seen map[*node]bool) (t 
 		// Resolve the left part of selector, then lookup the right part on it
 		var lt *itype
 
-		// If we are in a list of func parameters, and we are a selector on a binPkgT, but
-		// one of the other parameters has the same name as the pkg name, in the list of
-		// symbols we would find the other parameter instead of the pkg because it comes
-		// first when looking up in the stack of scopes. So in that case we force the
-		// lookup directly in the root scope to shortcircuit that issue.
-		var localScope *scope
-		localScope = sc
-		if n.anc != nil && len(n.anc.child) > 1 && n.anc.child[1] == n &&
-			// This check is weaker than what we actually want to know, i.e. whether
-			// n.anc.child[0] is a variable, but it seems at this point in the run we have no
-			// way of knowing that yet (typ is nil, so there's no typ.cat yet).
-			n.anc.child[0].kind == identExpr {
-			for {
-				if localScope.level == 0 {
-					break
-				}
-				localScope = localScope.anc
-			}
+		// Lookup the package symbol first if we are in a field expression as
+		// a previous parameter has the same name as the package, we need to
+		// prioritize the package type.
+		if n.anc.kind == fieldExpr {
+			lt = findPackageType(interp, sc, n.child[0])
 		}
-
-		if lt, err = nodeType2(interp, localScope, n.child[0], seen); err != nil {
-			return nil, err
+		if lt == nil {
+			// No package was found or we are not in a field expression, we are looking for a variable.
+			if lt, err = nodeType2(interp, sc, n.child[0], seen); err != nil {
+				return nil, err
+			}
 		}
 
 		if lt.incomplete {
@@ -925,6 +914,24 @@ func nodeType2(interp *Interpreter, sc *scope, n *node, seen map[*node]bool) (t 
 	}
 
 	return t, err
+}
+
+// findPackageType searches the top level scope for a package type.
+func findPackageType(interp *Interpreter, sc *scope, n *node) *itype {
+	// Find the root scope, the package symbols will exist there.
+	for {
+		if sc.level == 0 {
+			break
+		}
+		sc = sc.anc
+	}
+
+	baseName := filepath.Base(interp.fset.Position(n.pos).Filename)
+	sym, _, found := sc.lookup(filepath.Join(n.ident, baseName))
+	if !found || sym.typ == nil && sym.typ.cat != srcPkgT && sym.typ.cat != binPkgT {
+		return nil
+	}
+	return sym.typ
 }
 
 func isBuiltinCall(n *node, sc *scope) bool {

